server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -116,21 +122,27 @@ public void uploadBlobs(
     public void uploadBlobAsync(
         InputStream inputStream,
         Iterable<String> remotePath,
-        String blobName,
+        String fileName,
         WritePriority writePriority,
         ActionListener<Void> listener
     ) throws IOException {
         assert remotePath instanceof BlobPath;
         BlobPath blobPath = (BlobPath) remotePath;
         final BlobContainer blobContainer = blobStore.blobContainer(blobPath);
         if (blobContainer instanceof AsyncMultiStreamBlobContainer == false) {
-            blobContainer.writeBlob(blobName, inputStream, inputStream.available(), false);
+            blobContainer.writeBlob(fileName, inputStream, inputStream.available(), false);
             listener.onResponse(null);
             return;
         }
-        final String resourceDescription = "BlobStoreTransferService.uploadBlob(blob=\"" + blobName + "\")";
-        byte[] bytes = inputStream.readAllBytes();
-        try (IndexInput input = new ByteArrayIndexInput(resourceDescription, bytes)) {
+        final String resourceDescription = "BlobStoreTransferService.uploadBlob(blob=\"" + fileName + "\")";
```

Comment:
Aren't we reading the stream fully which will cause byte buffers to load the complete data in memory?

Reply:
Yes right. I think we would need to create a custom implementation of IndexInput to avoid read the stream fully.
Can we take that as a follow up item ?